lib/ansible/modules/cloud/azure/azure_rm_keyvault.py

```diff
@@ -0,0 +1,430 @@
+            location=dict(type='str'),
+            sku=dict(choices=['standard', 'premium'], default='standard', type='str'),
+            vault_uri=dict(type='str'),
+            tenant_id=dict(type='str'),
+            object_id=dict(type='str'),
+            application_id=dict(type='str'),
+            permissions=dict(choices=[
+                'Key, Secret, & Certificate Management',
+                'Key & Secret Management',
+                'Key Management',
+                'Secret Management',
+                'Certificate Management',
+                'SQL Server Connector',
+                'Azure Backup',
+                'Azure Data Lake Store'
+            ], default='Key, Secret, & Certificate Management', type='str')
+        )
+
+        # store the results of the module operation
+        self.results = dict(
+            changed=False,
+            state=dict()
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.state = None
+        self.enabled_for_deployment = None
+        self.enabled_for_disk_encryption = None
+        self.enabled_for_template_deployment = None
+        self.location = None
+        self.sku = None
+        self.vault_uri = None
+        self.tenant_id = None
+        self.object_id = None
+        self.application_id = None
+        self.permissions = None
+        self.tags = None
+
+        super(AzureRMKeyVault, self).__init__(self.module_arg_spec,
+                                              supports_check_mode=True,
+                                              supports_tags=True)
+
+    def exec_module(self, **kwargs):
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        self.results['check_mode'] = self.check_mode
+
+        # check if tenant or object id is passed via args if not pull from auth creds
+        if self.tenant_id is None:
+            if self.credentials['tenant'] is None:
+                self.fail("Please supply a tenant id")
+            else:
+                self.tenant_id = self.credentials['tenant']
+        if self.object_id is None:
+            if self.credentials['client_id'] is None:
+                self.fail("Please supply a client id")
+            else:
+                self.object_id = self.credentials['client_id']
+
+        try:
+            resource_group = self.get_resource_group(self.resource_group)
+        except CloudError:
+            self.fail('Unable to retrieve resource group')
+
+        self.location = self.location or resource_group.location
+
+        try:
+            vault = self.keyvault_client.vaults.get(self.resource_group, self.name)
+            exists = True
+        except CloudError as exc:
+            exists = False
+
+        if self.state == 'absent':
+            if exists:
+                if self.check_mode:
+                    self.results['changed'] = True
```

Comment:
You can move this out of both if statements and just set if before as both branches set it to true